cpp/oneapi/dal/backend/primitives/sparse_blas/set_csr_data_dpc.cpp

```diff
@@ -0,0 +1,130 @@
+                                     sparse_indexing_to_mkl(indexing),
+                                     const_cast<std::int64_t *>(row_offsets.get_data()),
+                                     const_cast<std::int64_t *>(column_indices.get_data()),
+                                     const_cast<Float *>(data.get_data()),
+                                     deps);
+}
+
+template <typename Float>
+sycl::event set_csr_data(sycl::queue &queue,
+                         sparse_matrix_handle &handle,
+                         const std::int64_t row_count,
+                         const std::int64_t column_count,
+                         dal::sparse_indexing indexing,
+                         const Float *data,
+                         const std::int64_t *column_indices,
+                         const std::int64_t *row_offsets,
+                         const event_vector &deps) {
+    ONEDAL_ASSERT(data);
+    ONEDAL_ASSERT(column_indices);
+    ONEDAL_ASSERT(row_offsets);
+    return mkl::sparse::set_csr_data(queue,
+                                     dal::detail::get_impl(handle).get(),
+                                     row_count,
+                                     column_count,
+                                     sparse_indexing_to_mkl(indexing),
+                                     const_cast<std::int64_t *>(row_offsets),
+                                     const_cast<std::int64_t *>(column_indices),
+                                     const_cast<Float *>(data),
+                                     deps);
+}
+
+sycl::event set_csr_data(sycl::queue &queue,
+                         sparse_matrix_handle &handle,
+                         const dal::csr_table &table,
+                         const event_vector &deps) {
+    ONEDAL_ASSERT(table.has_data());
+    data_type dtype = table.get_metadata().get_data_type(0);
+    ONEDAL_ASSERT(dtype == data_type::float32 || dtype == data_type::float64);
+
+    if (dtype == data_type::float32) {
+        return set_csr_data(queue,
+                            handle,
+                            table.get_row_count(),
+                            table.get_column_count(),
+                            table.get_indexing(),
+                            table.get_data<float>(),
+                            table.get_column_indices(),
+                            table.get_row_offsets(),
+                            deps);
+    }
+    else if (dtype == data_type::float64) {
+        return set_csr_data(queue,
+                            handle,
+                            table.get_row_count(),
+                            table.get_column_count(),
+                            table.get_indexing(),
+                            table.get_data<double>(),
+                            table.get_column_indices(),
+                            table.get_row_offsets(),
+                            deps);
+    }
+    return sycl::event();
```

Comment:
Should we throw exception here?

Reply:
It has an assertion upper in the code. I think the use of exceptions mechanism is limited in primitives (which are only for internal use) due to performance reasons.